satellite/marketing/offers.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package marketing
+
+import (
+	"context"
+	"time"
+
+	"github.com/zeebo/errs"
+)
+
+// OffersErr creates offer error class
+var OffersErr = errs.Class("offers error")
+
+// Offers holds information about offer
+type Offers interface {
+	ListAll(ctx context.Context) ([]Offer, error)
+	GetCurrentByType(ctx context.Context, offerType OfferType) (*Offer, error)
+	Create(ctx context.Context, offer *NewOffer) (*Offer, error)
+	Update(ctx context.Context, offer *UpdateOffer) error
```

Comment:
A single update somehow feels problematic here. You could get concurrent updates and probably would want to separate these... i.e. maybe having methods `Redeem`, `Renew` and `Finish` (or similar)?

Reply:
Wouldn't the lock file prevent that from happening?

Reply:
What do you mean by a lock file? If you mean a transactions, we need to reduce the number of transactions since it slows down the database.

Reply:
```golang
func (m *lockedOffers) Update(ctx context.Context, offer *marketing.UpdateOffer) error {
	m.Lock()
	defer m.Unlock()
	return m.db.Update(ctx, offer)
}
```
I thought this function in the `locked.go` would prevent concurrent update to the same data